### PR TITLE
Allow for f-contiguous 2D arrays in convex_hull_image

### PR DESCRIPTION
**Pull request by Dimchord**
## Description
This is the bug fix for issue #3758.

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/morphology/convex_hull.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def convex_hull_image(image, offset_coordinates=True, tolerance=1e-10):
     # the starting or ending pixel of a row or column.  This vastly
     # limits the number of coordinates to examine for the virtual hull.
     if ndim == 2:
-        coords = possible_hull(image.astype(np.uint8))
+        coords = possible_hull(np.ascontiguousarray(image.astype(np.uint8)))
```
Comment (hmaarrfk) [inline skimage/morphology/convex_hull.py]: I think maybe you should use the dtype parameter in the ascontiguousarray
https://docs.scipy.org/doc/numpy/reference/generated/numpy.ascontiguousarray.html#numpy-ascontiguousarray

```suggestion
        coords = possible_hull(np.ascontiguousarray(image, dtype=np.uint8))
```

**Comment by hmaarrfk** [comment]
Do you want to add a test for 3D images too? It seems that 2D images and 3D images go through different code paths.

**Comment by jni** [comment]
@hmaarrfk the 3D code path calls out to SciPy, so it's *probably* fine?

**Comment by hmaarrfk** [comment]
Yeah, I mostly don't know. I like to think that scikit-image is easy enough to mod that the next person who runs into the issue will be able to submit a small fix like this.

File: skimage/morphology/convex_hull.py (hunk 1 above)
Comment (Dimchord) [reply skimage/morphology/convex_hull.py]: I added this suggestion, as well as some tests for 3D arrays.

**Comment by hmaarrfk** [comment]
Thanks @Dimchord

**Pull request merged**